### PR TITLE
fix(email): enable multiple types of authentication

### LINKED ISSUE #65: new SMTP server doesn't support plain auth

### PR DESCRIPTION
**Pull request by sydneyli**
fixes #65

### CODE REVIEW

File: email.go [hunk 1]
```diff
@@ -36,6 +40,41 @@ Remember to read our guidelines (%s/policy-list) about the requirements your mai
...
 
+func makeEmailConfigFromEnv() (emailConfig, error) {
+	// create config
+	c := emailConfig{
+		username:           os.Getenv("SMTP_USERNAME"),
+		password:           os.Getenv("SMTP_PASSWORD"),
+		submissionHostname: os.Getenv("SMTP_ENDPOINT"),
+		port:               os.Getenv("SMTP_PORT"),
+		sender:             os.Getenv("SMTP_FROM_ADDRESS"),
+		website:            os.Getenv("FRONTEND_WEBSITE_LINK"),
```
Comment (wgreenberg) [inline email.go]: it doesn't seem like we validate `sender` or `website` are set; are they optional?

Comment (sydneyli) [reply email.go]: Not really. I'll set defaults for them if unspecified.

File: email.go [hunk 2]
```diff
@@ -36,6 +40,41 @@ Remember to read our guidelines (%s/policy-list) about the requirements your mai
...
+	err = client.StartTLS(&tls.Config{ServerName: c.submissionHostname})
+	if err != nil {
+		return c, fmt.Errorf("SMTP server doesn't support STARTTLS")
+	}
+	ok, auths := client.Extension("AUTH")
+	if !ok {
+		return c, fmt.Errorf("remote SMTP server doesn't support any authentication mechanisms")
+	}
+	if strings.Contains(auths, "PLAIN") {
+		c.auth = smtp.PlainAuth("", c.username, c.password, c.submissionHostname)
```
[the remark this answers is not in the document]
Comment (sydneyli) [reply email.go]: It's fine so long as the AUTH is performed post-STARTTLS.

[the remark this answers is not in the document]
Comment (sydneyli) [reply email.go]: ( to clarify, smtp.PlainAuth also only works over TLS: https://golang.org/pkg/net/smtp/#PlainAuth)

[the remark this answers is not in the document]
Comment (wgreenberg) [reply email.go]: aha, TIL!

**Comment by sydneyli** [comment]
@wgreenberg added a composite errors class that helped out. Not sure if it's "go-like," but I think it's clean-- errors are being caught in `log.Fatalf` in main.go now, which is nice :)

**Comment by sydneyli** [comment]
(got an OK from @wgreenberg out-of-band)

**Pull request merged**